### PR TITLE
[SPARK-28835][SQL][TEST] Add TPCDSSchema trait

### PR DESCRIPTION
**Pull request by dbaliafroozeh**
### What changes were proposed in this pull request?
This PR extracts the schema information of TPCDS tables into a separate class called `TPCDSSchema` which can be reused for other testing purposes

### How was this patch tested?
This PR is only a refactoring for tests and passes existing tests

### CODE REVIEW

**Comment by hvanhovell** [comment]
ok to test

**Review by hvanhovell — APPROVED** [review]
LGTM

**Review by gengliangwang — APPROVED** [review]
LGTM. It would be better to add [TEST] to the PR title.

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        |`web_street_name` STRING, `web_street_type` STRING, `web_suite_number` STRING,
+        |`web_city` STRING, `web_county` STRING, `web_state` STRING, `web_zip` STRING,
+        |`web_country` STRING, `web_gmt_offset` DECIMAL(5,2), `web_tax_percentage` DECIMAL(5,2)
+        |""".stripMargin
+  )
+
+  val tableNames: Iterable[String] = tableColumns.keys
+
+  def createTable(
+    spark: SparkSession,
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala]: Could you add two more space for indentation? According to [Apache Spark contribution guide](https://spark.apache.org/contributing.html), you can follow the following style.
- https://github.com/databricks/scala-style-guide#indent

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql
+
+trait TPCDSSchema {
+
+  private val tableColumns = Map(
+    "call_center" ->
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala]: @dbaliafroozeh . I'm just wondering  you changed the order of tables from the original code? For me,
1. This is not consistent with the original code (It starts with `catalog_page` table.)
    - This is the best for review.
2. This is not consistent with TPCDS spec document. (Usually, it starts with `Fact Table Definition`: `store_sales` table)
    - This is the best to compare with TPCDS spec doc.
3. This is not alphabetic.
    - This is the best for human readability and to lookup on GitHub.

While you are here, could you choose one of the above and update this PR?

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala [hunk 3]
```diff
@@ -0,0 +1,259 @@
...
+        |`cc_call_center_sk` INT, `cc_call_center_id` STRING, `cc_rec_start_date` DATE,
+        |`cc_rec_end_date` DATE, `cc_closed_date_sk` INT, `cc_open_date_sk` INT, `cc_name` STRING,
+        |`cc_class` STRING, `cc_employees` INT, `cc_sq_ft` INT, `cc_hours` STRING,
+        |`cc_manager` STRING, `cc_mkt_id` INT, `cc_mkt_class` STRING, `cc_mkt_desc` STRING,
+        |`cc_market_manager` STRING, `cc_division` INT, `cc_division_name` STRING,
+        |`cc_company` INT, `cc_company_name` STRING, `cc_street_number` STRING,
+        |`cc_street_name` STRING, `cc_street_type` STRING, `cc_suite_number` STRING,
+        |`cc_city` STRING, `cc_county` STRING, `cc_state` STRING, `cc_zip` STRING,
+        |`cc_country` STRING, `cc_gmt_offset` DECIMAL(5,2), `cc_tax_percentage` DECIMAL(5,2)
+        |""".stripMargin,
```
Comment (dongjoon-hyun) [inline sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala]: Please match the starting and ending indentation of `"""`. You can see it from [the deleted code](https://github.com/apache/spark/pull/25535/files#diff-38fa80d1dc9860f07e135dd02d259269L39).

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
Thank you for your contribution, @dbaliafroozeh .
I left a few minor comments. Could you update the PR accordingly, please?

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala (hunk 2 above)
Comment (dbaliafroozeh) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala]: I have updated the list of tables according to the TPCDS spec (option 2). In addition of being consistent with the spec, having fact tables `_sales` and `_return` next to each other looked most logical to me.

**Comment by dbaliafroozeh** [comment]
retest this please

**Comment by dbaliafroozeh** [comment]
Jenkins, retest this please

**Comment by maropu** [comment]
retest this please

**Comment by maropu** [comment]
Any plan to reuse this traint in your following activities?

File: sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala (hunk 2 above)
Comment (dongjoon-hyun) [reply sql/core/src/test/scala/org/apache/spark/sql/TPCDSSchema.scala]: Could you include the link to the PR description? It's always good to explain the way of your thought.

**Review by dongjoon-hyun — APPROVED** [review]
+1, LGTM. Thank you, @dbaliafroozeh , @hvanhovell , @gengliangwang , @maropu !
Merged to master.

cc @gatorsmile

**Comment by dongjoon-hyun** [comment]
Oops. @dbaliafroozeh . It seems that you use a wrong JIRA id here.

**Comment by dongjoon-hyun** [comment]
Since this is merged already, I'll switch SPARK-28835 and SPARK-28836

**Comment by HyukjinKwon** [comment]
@dbaliafroozeh, can you make sure the PR description and JIRA description are reasonable next time? I put some efforts to make it consistent at https://github.com/apache/spark/commit/0ea8db9fd3d882140d8fa305dd69fc94db62cf8f but seems those were intentionally removed out here.

> ... which can be reused for other testing purposes

I would do such stuff when we actually reuse rather then over-refactoring.

Also, I closed your JIRA before because I didn't know what the JIRA meant. Please just don't copy and paste the description into JIRA next time.

**Comment by HyukjinKwon** [comment]
@dbaliafroozeh, let's make the PR description and JIRA description reasonable next time. I put some efforts to make it consistent and manageable at https://github.com/apache/spark/commit/0ea8db9fd3d882140d8fa305dd69fc94db62cf8f 

Also, I mistakenly closed your JIRA because I misunderstood the JIRA given the JIRA description.

**Comment by dbaliafroozeh** [comment]
@dongjoon-hyun Sorry for wrong Jira id, and thanks for fixing it.

**Pull request closed without merging**